### PR TITLE
Re2parser and re2 library

### PR DESCRIPTION
**Pull request by martinhruska**
This pull requests:

1. Adds RE2 library to the 3rd party libraries
2. Adds parser for transforming RE2 regexps to our NFA. It was implemented by Michal Horky, a student of Lukas.
3. Adds tests for the mentioned parser.

The pull request is based on integration of RE2 parser and library which Michal Horky did to the automata library devoloped by Lukas and Juraj.

### CODE REVIEW

**Comment by martinhruska** [comment]
I cleaned up interface and removed re2 dependencies.

**Comment by ondrik** [comment]
seems good to me from a bird's-eye view

File: src/re2parser.cc [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+        }
+        };
+}
+
+ /**
+ * The main method, it creates NFA from regex
+ * @param pattern regex as string
+ * @return vata2::Nfa::Nfa corresponding to pattern
+ */
+Vata2::Nfa::Nfa Vata2::RE2Parser::create_nfa(const std::string& pattern) {
```
Comment (tfiedor) [inline src/re2parser.cc:513]: I would suggest to create paired function with signature `void create_nfa(Nfa*, string)`

**Review by tfiedor — APPROVED** [review]
The only change I would suggest is to pair the `create_nfa` function with other signature (it is easier for python binding, and conforms to the style of the library.

**Comment by ondrik** [comment]
maybe also check the pointer is not null...

**Comment by martinhruska** [comment]
null pointer check added.

**Pull request merged**